make/protogen.mk

```diff
@@ -168,61 +159,93 @@ printprotos:
 
-$(PROTOC_GEN_SWAGGER): $(MODFILE_DIR)/github.com/grpc-ecosystem/grpc-gateway/UPDATE_CHECK $(PROTO_GOBIN)
+PROTOC_GEN_OPENAPIV2 := $(PROTO_GOBIN)/protoc-gen-openapiv2
+$(PROTOC_GEN_OPENAPIV2): $(MODFILE_DIR)/github.com/grpc-ecosystem/grpc-gateway/v2/UPDATE_CHECK $(PROTO_GOBIN)
 	@echo "+ $@"
-	$(SILENT)GOBIN=$(PROTO_GOBIN) $(GOINSTALL) github.com/grpc-ecosystem/grpc-gateway/protoc-gen-swagger
+	$(SILENT)GOBIN=$(PROTO_GOBIN) $(GOINSTALL) github.com/grpc-ecosystem/grpc-gateway/v2/protoc-gen-openapiv2
 
 $(GENERATED_DOC_PATH):
 	@echo "+ $@"
 	$(SILENT)mkdir -p $(GENERATED_DOC_PATH)
 
-# Generate all of the proto messages and gRPC services with one invocation of
-# protoc when any of the .pb.go sources don't exist or when any of the .proto
-# files change.
-$(GENERATED_BASE_PATH)/%.pb.go: $(PROTO_BASE_PATH)/%.proto $(PROTO_DEPS) $(PROTOC_GEN_GRPC_GATEWAY) $(PROTOC_GEN_GO_BIN) $(ALL_PROTOS)
+# Generate all of the proto messages with one invocation of protoc when
+# any of the .pb.go sources don't exist or when any of the .proto files change.
+$(GENERATED_BASE_PATH)/%.pb.go: $(PROTO_BASE_PATH)/%.proto $(PROTO_DEPS) $(PROTOC_GEN_GO_BIN) $(ALL_PROTOS)
+	@echo "+ $@"
+	$(SILENT)mkdir -p $(dir $@)
+	$(SILENT)PATH=$(PROTO_GOBIN) $(PROTOC) \
+		-I$(PROTOC_INCLUDES) \
+		-I$(GOOGLE_API_INCLUDES) \
+		--proto_path=$(PROTO_BASE_PATH) \
+		--plugin protoc-gen-go="${PROTOC_GEN_GO_BIN}" \
+		--go_out=$(M_ARGS_STR:%=%,)module=github.com/stackrox/scanner/generated:$(GENERATED_BASE_PATH) \
+		$(dir $<)/*.proto
+
+# Generate all of the vtproto extensions with one invocation of protoc when
+# any of the *_vtproto.pb.go sources don't exist or when any of the .proto files change.
+$(GENERATED_BASE_PATH)/%_vtproto.pb.go: $(PROTO_BASE_PATH)/%.proto $(PROTO_DEPS) $(PROTOC_GEN_GO_VTPROTO_BIN) $(ALL_PROTOS)
+	@echo "+ $@"
+	$(SILENT)mkdir -p $(dir $@)
+	$(SILENT)PATH=$(PROTO_GOBIN) $(PROTOC) \
+		-I$(PROTOC_INCLUDES) \
+		-I$(GOOGLE_API_INCLUDES) \
+		--proto_path=$(PROTO_BASE_PATH) \
+		--plugin protoc-gen-go-vtproto="${PROTOC_GEN_GO_VTPROTO_BIN}" \
+		--go-vtproto_opt=features=marshal+unmarshal+size+equal+clone+pool \
+		--go-vtproto_out=$(M_ARGS_STR:%=%,)module=github.com/stackrox/scanner/generated:$(GENERATED_BASE_PATH) \
+		$(dir $<)/*.proto
+
+# Generate all of the gRPC bindings with one invocation of protoc when
+# any of the *_grpc.pb.go sources don't exist or when any of the .proto files change.
+$(GENERATED_BASE_PATH)/%_grpc.pb.go: $(PROTO_BASE_PATH)/%.proto $(PROTO_DEPS) $(PROTOC_GEN_GO_GRPC) $(ALL_PROTOS)
 	@echo "+ $@"
 	$(SILENT)mkdir -p $(dir $@)
 	$(SILENT)PATH=$(PROTO_GOBIN) $(PROTOC) \
-		-I$(GOGO_DIR) \
 		-I$(PROTOC_INCLUDES) \
-		-I$(GRPC_GATEWAY_DIR)/third_party/googleapis \
+		-I$(GOOGLE_API_INCLUDES) \
 		--proto_path=$(PROTO_BASE_PATH) \
-		--gofast_out=$(GOGO_M_STR:%=%,)$(M_ARGS_STR:%=%,)plugins=grpc:$(GENERATED_BASE_PATH) \
+		--plugin protoc-gen-go-grpc="${PROTOC_GEN_GO_GRPC}" \
```

Comment:
Since we pass full path to plugin binary do we need to  change `$PATH` in 218?

Reply:
We can do that. I would do that as a follow-up and do a cleanup of `make/protogen.mk`. All swagger.json targets can be removed.